blueflood-elasticsearch/src/main/java/com/rackspacecloud/blueflood/io/ElasticsearchRestHelper.java

```diff
@@ -469,51 +485,65 @@ private String getUnit(Metric metric) {
         return metric.getUnit();
     }
 
-    public int index(String urlFormat, String bulkString) throws IOException {
-        String tempUrl = String.format(urlFormat, getBaseUrl());
-        HttpEntity entity = new NStringEntity(bulkString, ContentType.APPLICATION_JSON);
-        int statusCode = 0;
-
-        Queue<String> callQ = new LinkedList<>();
-        callQ.add(tempUrl);
-        int callCount = 0;
-        while(!callQ.isEmpty() && callCount < MAX_CALL_COUNT) {
-            callCount++;
-            String url = callQ.remove();
-            logger.debug("Using url [{}]", url);
-            HttpPost httpPost = new HttpPost(url);
-            httpPost.setHeaders(getHeaders());
-            httpPost.setEntity(entity);
-
-            CloseableHttpResponse response = null;
+    public ListenableFuture<Boolean> index(final String urlFormat, final String bulkString) throws IOException {
+        int remainingCapacityOfTheBlockingQueue = threadPoolManager.remainingCapacityOfTheQueue();
+        logger.debug("Remaining capacity of ES blocking queue: [" + remainingCapacityOfTheBlockingQueue + "]");
 
-            try {
-                response = closeableHttpClient.execute(httpPost);
-                statusCode = response.getStatusLine().getStatusCode();
-                String str = EntityUtils.toString(response.getEntity());
-                EntityUtils.consume(response.getEntity());
+        if(threadPoolManager.remainingCapacityOfTheQueue() == 0) {
+            logger.warn("Remaining capacity of ES blocking queue: [" + remainingCapacityOfTheBlockingQueue + "]");
+        }
 
-                if (statusCode != HttpStatus.SC_OK && statusCode != HttpStatus.SC_CREATED) {
-                    logger.error("index method failed with status code: {} and error: {}", statusCode, str);
-                }
-            } catch (Exception e) {
-                if(response == null){
-                    logger.error("index method failed with message: {}", e.getMessage());
-                    url = String.format(urlFormat, getBaseUrl());
-                    callQ.add(url);
-                }
-                else {
-                    logger.error("index method failed with status code: {} and exception message: {}",
-                            statusCode, e.getMessage());
-                }
-            } finally {
-                if(response != null) {
-                    response.close();
+        return threadPoolManager.listeningExecutorService.submit(() -> {
+            String tempUrl = String.format(urlFormat, getBaseUrl());
+            HttpEntity entity = new NStringEntity(bulkString, ContentType.APPLICATION_JSON);
+            int statusCode = 0;
+
+            Queue<String> callQ = new LinkedList<>();
```

Comment:
oh.. this is just implementation of queueing the different endpoints so that if one call fails it will try other endpoint. I will add this as the comment to the code.